src/character.cpp

```diff
@@ -1949,7 +1949,8 @@ bool Character::pour_into( item &container, item &liquid )
     const long amount = container.get_remaining_capacity_for_liquid( liquid, *this, &err );
 
     if( !err.empty() ) {
-        add_msg_if_player( m_bad, err.c_str() );
+        add_msg_if_player( m_bad, string_format( "You can't pour %1$s in your %2$s: it %3$s.",
```

Comment:
Please let `add_msg_if_player` do the formatting. You're currently inviting UB, for example when one of the `tname` function calls returns a string that contains a format specifier. Those would be forwarded to `add_msg_if_player`, which would therefor expect further arguments.